packages/cli/src/cli/config/action.bundle.assets.ts

```diff
@@ -0,0 +1,115 @@
+import { CommandLineAction, CommandLineStringParameter } from '@rushstack/ts-command-line';
+import { SourceMemory } from '@chunkd/core';
+import { CotarIndexBinary, CotarIndexBuilder, CotarIndexOptions, TarReader } from '@cotar/core';
+import { LogConfig, LogType } from '@basemaps/shared';
+import { promises as fs } from 'fs';
+import { TarBuilder } from '@cotar/tar';
+import { fsa } from '@chunkd/fs';
+
+const Packing = 25; // Packing factor for the hash map
+const MaxSearch = 50; // Max search factor
+
+export class CommandBundleAssets extends CommandLineAction {
+  assets: CommandLineStringParameter;
+  output: CommandLineStringParameter;
+
+  public constructor() {
+    super({
+      actionName: 'bundle-assets',
+      summary: 'Cli tool to create cotar file for the config assets',
+      documentation: 'Create a cotar file from a directory',
+    });
+  }
+
+  protected onDefineParameters(): void {
+    this.assets = this.defineStringParameter({
+      argumentName: 'ASSETS',
+      parameterLongName: '--assets',
+      description: 'Paths to the input assets files, must contain assets/sprites/ and assets/fonts.',
+    });
+
+    this.output = this.defineStringParameter({
+      argumentName: 'OUTPUT',
+      parameterLongName: '--output',
+      description: 'Paths of the output co tar file.',
+    });
+  }
+
+  protected async onExecute(): Promise<void> {
+    const logger = LogConfig.get();
+    const assets = this.assets.value;
+    const output = this.output.value;
+    if (assets == null || output == null) throw new Error('Please provide a input or ouput path');
+    if (!output.endsWith('.tar.co')) throw new Error(`Invalid output, needs to be .tar.co :"${output}"`);
+
+    logger.info({ indput: assets }, 'BundleAssets:Start');
+    const tarFile = await this.buildTar(assets, output, logger);
+    const cotarFile = await this.buildTarCo(tarFile, output, logger);
+    logger.info({ output: cotarFile }, 'BundleAssets:Finish');
+  }
+
+  async buildTar(input: string, output: string, logger: LogType): Promise<string> {
+    const startTime = Date.now();
+    const outputTar = output.split('.co')[0];
+
+    const tarBuilder = new TarBuilder(outputTar);
+
+    const files = await fsa.toArray(fsa.list(input));
+    // Ensure files are put into the same order
+    files.sort((a, b) => a.localeCompare(b));
+    logger.info({ output: outputTar, files: files.length }, 'Tar:Create');
+
+    for (const file of files) {
```

Comment:
I think the best option would be to use something like
```typescript
const basePath = path.resolve(input)
filePath = file.replace(basePath, '')
```

Reply:
Fixed. Thanks you.

Reply:
I didnt actually check this, does this actually give us the correct tar output?

Reply:
Leading / removed now. Thanks.